### PR TITLE
docs: elaborate on Scylla admin REST API

### PR DESCRIPTION
**Pull request by tzach**
Adding more information and context on the REST API, including a docker example and a swagger screen capture.

Rander page below:
image

### CODE REVIEW

**Comment by tzach** [comment]
@dgarcia360 any idea why I get `operating-scylla/rest.rst:3: WARNING: Title level inconsistent` ?

The title is *not* updated in this PR
```

======
 REST
======
```

**Comment by dgarcia360** [comment]
Note that this file is included in `docs/operating-scylla/admin.rst`:

image

And, this PR edits the heading syntax to an H1:

image

Sphinx complains because detects an H1 under an H2. To solve this issue, add back the previous heading syntax:

```
REST
----
```

File: docs/operating-scylla/rest.rst [hunk 1]
```diff
@@ -1,8 +1,27 @@
 
 REST
-----
+------
 
-For each JMX operation, attribute get and set, Scylla exposes a matching REST API. You can interact with the REST API using :code:`curl` or using the Swagger UI available at :code:`your-ip:10000/ui`
+Scylla exposes a REST API to retrieve administrative information from a node and execute administrative operations.
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:5]:
```suggestion
ScyllaDB exposes a REST API to retrieve administrative information from a node and execute administrative operations.```

Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:5]:
```suggestion
ScyllaDB exposes a REST API to retrieve administrative information from a node and execute administrative operations.
```

File: docs/operating-scylla/rest.rst [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+Examples are configuration status and updates, cluster-level information, and more.
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:6]:
```suggestion
For example, it allows you to check or update configuration, retrieve cluster-level information, and more.
```

File: docs/operating-scylla/rest.rst [hunk 3]
```diff
@@ -1,8 +1,27 @@
...
 
+:doc:`Nodetool </operating-scylla/nodetool>` CLI tool interacts, using JMX, with a *scylla-jmx* process, which in turn, uses the REST API to interact with the Scylla process.
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:8]:
```suggestion
The :doc:`nodetool </operating-scylla/nodetool>` CLI tool interacts with a *scylla-jmx* process using JMX. The process, in turn, uses the REST API to interact with the ScyllaDB process.
```

File: docs/operating-scylla/rest.rst [hunk 4]
```diff
@@ -1,8 +1,27 @@
...
-----
+------
 
-For each JMX operation, attribute get and set, Scylla exposes a matching REST API. You can interact with the REST API using :code:`curl` or using the Swagger UI available at :code:`your-ip:10000/ui`
+Scylla exposes a REST API to retrieve administrative information from a node and execute administrative operations.
+Examples are configuration status and updates, cluster-level information, and more.
 
+:doc:`Nodetool </operating-scylla/nodetool>` CLI tool interacts, using JMX, with a *scylla-jmx* process, which in turn, uses the REST API to interact with the Scylla process.     
 
+You can interact with the REST API directly using :code:`curl` or using the Swagger UI available at :code:`your-ip:10000/ui`
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:10]:
```suggestion
You can interact with the REST API directly using :code:`curl` or using the Swagger UI available at :code:`your-ip:10000/ui`.
```

File: docs/operating-scylla/rest.rst [hunk 5]
```diff
@@ -1,8 +1,27 @@
...
 
+For example, using the Swagger UI with Scylla Docker:
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:12]:
```suggestion
The following example shows using the Swagger UI with the Scylla Docker image.
```

File: docs/operating-scylla/rest.rst [hunk 6]
```diff
@@ -1,8 +1,27 @@
...
+
+.. code:: sh
+
+  docker run --name some-scylla -p 10000:10000 -d scylladb/scylla:5.0.4 --api-address 0.0.0.0
+
+and visting http://localhost:10000/ui/
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:18]:
```suggestion
Then visit http://localhost:10000/ui/.
```

File: docs/operating-scylla/rest.rst [hunk 7]
```diff
@@ -1,8 +1,27 @@
...
+
+.. warning::
+
+  Do not expose the REST API externally, as done in the example above, for production systems.
+  It will allow external access to administration-level information and operations on your cluster!
+  By default, the API is bound to an internal IP (127.0.0.1), blocked for external connections.
+
```
Comment (annastuchlik) [inline docs/operating-scylla/rest.rst:25]:
```suggestion
```

**Review by annastuchlik — CHANGES_REQUESTED** [review]
I've left some suggestions, mostly about rewording, and I moved the warning so that it doesn't come between the command and the screenshot.

File: docs/operating-scylla/rest.rst (hunk 5 above)
Comment (tzach) [reply docs/operating-scylla/rest.rst:12]: @annastuchlik its hard for me to parse this suggestion

**Comment by annastuchlik** [comment]
@tzach This is the feedback I previously left as a suggestion - regarding the warning:

- The warning should be moved between "_The following example shows using the Swagger UI with the Scylla Docker image._" and "_Run the Docker command line:_".
- I would rephrase the warning for better readability:

.. warning::
   The following example is only for demonstration purposes, as it exposes the REST API externally.
   
   **Do not expose the REST API externally for production systems.** It will allow external access to administration-level information and operations on your cluster!
  
   By default, the API is bound to an internal IP (127.0.0.1), blocked for external connections.

**Review by annastuchlik — APPROVED** [review]
LGTM

**Comment by tzach** [comment]
@nyh can you please review / merge

**Comment by annastuchlik** [comment]
@amnonh Could you review this PR so that it can be merged?

@tzach Should it be only merged to master or backported to branch-5.2? I'm asking, because the follow-up PR https://github.com/scylladb/scylladb/issues/11999 should be backported.

Anyway, we should merge this PR first and then continue with https://github.com/scylladb/scylladb/issues/11999.

**Comment by tzach** [comment]
No need to backport.

**Comment by annastuchlik** [comment]
@amnonh @nyh Could you review this PR?

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please merge.

**Comment by denesb** [comment]
There are conflicts, please rebase.

**Comment by annastuchlik** [comment]
@tzach Is it possible for you to rebase and resolve the conflicts or do you want me to open a V2 PR?

**Comment by tzach** [comment]
Please send V2

**Comment by annastuchlik** [comment]
@tzach Not yet, but I have it queued and do it shortly.

**Comment by annastuchlik** [comment]
Closing in favor of V2: https://github.com/scylladb/scylladb/pull/13331
